packages/flutter_tools/lib/src/template.dart

```diff
@@ -43,20 +66,23 @@ class Template {
        _templateManifest = templateManifest {
     _templateFilePaths = <String, String>{};
 
-    if (!templateSource.existsSync()) {
+    if (!templateSources.every((Directory directory) => directory.existsSync())) {
       return;
```

Comment:
is this a silent failure? or? i'm not sure what this if statement is for. when would we go down this branch? might be good to add a comment giving the answer

Reply:
I agree this should probably be a `throwToolExit` if a template source directory does not exist.

@jonahwilliams Do you happen to know why this code branch does not throw an exception? It didn't before, so I assumed that silent return handled some use-case I hadn't discovered.

Reply:
throwToolExit SGTM. The exists sync check was added in 2016, so we might be past that now...

Reply:
Thanks! I've added a throwToolExit and corresponding test.